docs/sources/k6/next/misc/versioning-and-stability-guarantees.md

```diff
@@ -0,0 +1,265 @@
+
+Because we follow the [semver 2.0.0](https://semver.org/spec/v2.0.0.html) versioning scheme, we work under the assumption breaking changes will always lead to a new major version. When the k6 team decides to introduce a breaking change, it will automatically define the need for a new major version.
+
+When k6 releases a new major version, we are expected to be explicit, clearly communicate the breaking changes, and provide instructions for migrating users.
+
+There are two types of breaking changes:
+
+1. Removal
+2. Update
+
+#### Steps to remove an API or a feature
+
+1. Deprecate a current API or a feature by following the dedicated policy.
+2. If necessary, add an alternative solution. Ideally, this would be an experimental version within a minor release within the current maintained state. Alternatively, it might be available in a Preview release.
+3. Provide documentation for the migration path.
+4. Proceed with removing the API and releasing a new major version.
+
+#### Steps to update an API or a feature
+
+1. Apply the changes, then publish them in the next major version. If the change is heavy, it might be convenient to make it available in a Preview release that can support the migration path until a version is globally available.
+2. Provide documentation for the migration path.
+
+### Deprecation
+
+When the k6 team intends to remove an API or a feature, it will be marked as *deprecated*. This occurs when an API becomes obsolete, superseded by another API, or otherwise discontinued for business or technological reasons.
+Deprecated APIs remain available through their deprecated phase, which guarantees they will remain available at least throughout the entire life cycle of the current major version.
+
+A deprecation has several layers of communication necessary to be implemented and respected before it can be officially applied:
+
+1. Release notes and changelogs
+2. Official documentation
+3. When the user directly interacts with the feature
+4. An alternative already available is recommended, but it shouldn’t be considered mandatory.
+### Migration
+
+When a breaking change is introduced, or an API is superseded, then documentation for a migration guide must be provided to users to help them understand how to move from version to version.
+The migration path must be available in the official documentation, and it is recommended to be mentioned in all other communication channels.
+## Support and development policies
+
+We define k6 support and development policies according to four states where a k6 version can be implemented.
+
+### The states of k6 releases
+
+At any given time, there is a:
+
+* **The under-development** version is the next upcoming version, k6. Note that this version might either start a new major version (*e.g. developing v3.0.0, while the latest release is v2.8.2),* or it might be a new minor or patch iteration over the latest release.
```

Comment:
```suggestion
* **The under-development** version is the next upcoming version of k6. Note that this version might either start a new major version, for example, developing v3.0.0 while the latest release is v2.8.2, or it might be a new minor or patch iteration over the latest release.
```